### PR TITLE
arcgis-rest-common-types -- Added WebMap interfaces

### PR DESCRIPTION
**Pull request by JeffJacobson**
feat(arcgis-rest-common-types): Added [WebMap Specification](https://developers.arcgis.com/web-map-specification/) interfaces and types.

build(arcgis-rest-common-types): simplified build
* Activated *emitDeclarationOnly* compiler option, which eliminates the need for the
subsequent step for deleting compiled (empty) JavaScript files.

### CODE REVIEW

File: packages/arcgis-rest-common-types/package.json [hunk 1]
```diff
@@ -6,13 +6,12 @@
   "author": "",
   "license": "Apache-2.0",
   "files": [
-    "dist/types/index.d.ts"
+    "dist/types/*.d.ts"
```
Comment (jgravois) [inline packages/arcgis-rest-common-types/package.json]: i didn't know you could do this. very cool!

**Comment by jgravois** [comment]
ugh. travis is doing the same thing as in https://github.com/Esri/hub.js/pull/27#issuecomment-388949335

we'll figure it out and get this landed soon. thanks for the contribution @JeffJacobson!

File: packages/arcgis-rest-common-types/src/webmap.ts [hunk 2]
```diff
@@ -0,0 +1,1225 @@
...
+  /** Indicates whether to display the time slider (if accessible). */
+  displayTimeSlider?: true;
+  /** Indicates whether to display a legend on all slides. */
+  showLegend?: true;
+  /** Number of seconds to show slide before proceeding to the next slide. */
+  slideAdvancementInterval?: number;
+  /** Array of slide objects. */
+  slides?: ISlide[];
+  /** Indicates whether to use the time extent (if applicable) of a slide. */
+  useTimeExtentOfSlide?: true;
```
Comment (jgravois) [inline packages/arcgis-rest-common-types/src/webmap.ts]: ~~`true`~~ > `boolean`? (there are a handful like this.)

File: packages/arcgis-rest-common-types/src/webmap.ts [hunk 3]
```diff
@@ -0,0 +1,1225 @@
...
+}
+
+export interface ISupportsTime {
+  /** This property is applicable to layers that support time. If 'true', timeAnimation is enabled. */
+  timeAnimation?: boolean;
+}
+
+export interface IBingLayer extends ILayer {
+  /** String value that unlocks the use of Bing layers in a web map. */
+  bingKey?: any;
```
Comment (jgravois) [inline packages/arcgis-rest-common-types/src/webmap.ts]: `string`?

File: packages/arcgis-rest-common-types/src/webmap.ts [hunk 4]
```diff
@@ -0,0 +1,1225 @@
...
+  /** Boolean value indicating whether the Bing key can be shared to the public. */
+  canShareBingPublic?: any;
```
Comment (jgravois) [inline packages/arcgis-rest-common-types/src/webmap.ts]: `boolean`?

File: packages/arcgis-rest-common-types/src/webmap.ts [hunk 5]
```diff
@@ -0,0 +1,1225 @@
...
+  /** String indicating the layer type. */
+  layerType: "BingMapsAerial" | "BingMapsRoad" | "BingMapsHybrid";
+  /** A string value representing the URL to the Portal/organization Self resource. Calls should be made to this property to retrieve the Bing key. If the key is not made accessible to the public or if canShareBingPublic is false, any web maps using Bing layers will not work. */
+  portalUrl?: any;
+}
+
+export interface IImageServiceLayer extends ILayer, ISupportsTime {
+  /** An array of bandIds that are visible, can specify bands to export or rearrange band order(from image service). */
+  bandIds?: any;
```
Comment (jgravois) [inline packages/arcgis-rest-common-types/src/webmap.ts]: `number[]`?

File: packages/arcgis-rest-common-types/src/webmap.ts [hunk 6]
```diff
@@ -0,0 +1,1225 @@
...
+  /** Controls how much loss the image will be subjected to by the compression algorithm (from image service). */
+  compressionQuality?: any;
+  /** Stores interactive filters. */
+  definitionEditor?: any;
+  /** Boolean property indicating whether to ignore popups defined by the service item. */
+  disablePopup?: any;
```
Comment (jgravois) [inline packages/arcgis-rest-common-types/src/webmap.ts]: `boolean`

File: packages/arcgis-rest-common-types/src/webmap.ts [hunk 7]
```diff
@@ -0,0 +1,1225 @@
...
+  popupInfo?: IPopupInfo;
+  /** Number describing the size of the tile. */
+  symbolTileSize?: any;
+}
+
+export interface IDefinitionParameter {
+  /** The default value that is automatically given is nothing is provided. */
+  defaultValue?: number | string;
+  /** A string value representing the name of the field to query. */
+  fieldName?: any;
```
Comment (jgravois) [inline packages/arcgis-rest-common-types/src/webmap.ts]: `string`

File: packages/arcgis-rest-common-types/src/webmap.ts [hunk 8]
```diff
@@ -0,0 +1,1225 @@
...
+  /** A featureSet object containing the geometry and attributes of the features in the layer. Used with feature collections only. */
+  featureSet?: IFeatureSet;
+  /** Information about each field in a layer. Used with feature collections. */
+  field?: IField[];
+  /** A string indicating the index position of the layer in the map service or feature collection. */
+  id?: string;
+  /** The layerDefinition object defines the attribute schema and drawing information for the layer. */
+  layerDefinition?: any;
+  /** A string URL to a service that should be used for all queries against the layer. Used with hosted tiled map services on ArcGIS Online when there is an associated feature service that allows for queries. */
+  layerUrl?: any;
```
Comment (jgravois) [inline packages/arcgis-rest-common-types/src/webmap.ts]: `string`

File: packages/arcgis-rest-common-types/src/webmap.ts [hunk 9]
```diff
@@ -0,0 +1,1225 @@
...
+  /** A popupInfo object defining the popup window content for the layer. */
+  popupInfo?: IPopupInfo;
+  /** Indicates whether to allow map authors the ability to control what layers should be shown in a client's legend. */
+  showLegend?: any;
+  /** Array of numeric values indicating the layer index for layers nested directly under the parent layer. */
+  subLayer?: any;
+  /** If the layer is a parent layer, it will have one or more sub layers included in an array. */
+  subLayerIds?: any;
+  /** A user-friendly string title for the layer that can be used in a table of contents. */
+  title?: any;
```
Comment (jgravois) [inline packages/arcgis-rest-common-types/src/webmap.ts]: `string`

File: packages/arcgis-rest-common-types/src/webmap.ts [hunk 10]
```diff
@@ -0,0 +1,1225 @@
...
+}
+
+export interface IMapServiceLayer extends ILayer, ISupportsTime {
+  /** This is applicable if used as a baseMapLayer. A boolean value indicating whether or not the baseMapLayer draws on top (true) of other layers, including operationalLayers , or below (false). */
+  isReference?: any;
```
Comment (jgravois) [inline packages/arcgis-rest-common-types/src/webmap.ts]: `boolean`

File: packages/arcgis-rest-common-types/src/webmap.ts [hunk 11]
```diff
@@ -0,0 +1,1225 @@
...
+
+export interface IDrawingInfo {
+  /** Only used for feature collections with a renderer. The feature's symbol is defined by the layer's renderer. */
+  fixedSymbols?: any;
+  /** An object defining the properties used for labeling the layer. */
+  labelingInfo?: any;
+  /** An object defined which provides the symbology for the layer. */
+  renderer?: any;
+  /** Boolean property indicating whether symbols should stay the same size in screen units as you zoom in. A value of true means the symbols stay the same size in screen units regardless of the map scale. */
+  scaleSymbols?: any;
```
Comment (jgravois) [inline packages/arcgis-rest-common-types/src/webmap.ts]: `boolean`

File: packages/arcgis-rest-common-types/src/webmap.ts [hunk 12]
```diff
@@ -0,0 +1,1225 @@
...
+  /** Defines whether a labels should be shown or not. This is only valid for sublayers. */
+  showLabels?: any;
```
Comment (jgravois) [inline packages/arcgis-rest-common-types/src/webmap.ts]: `boolean`

File: packages/arcgis-rest-common-types/src/webmap.ts [hunk 13]
```diff
@@ -0,0 +1,1225 @@
...
+  /** Number value ranging between 0 (no transparency) to 100 (completely transparent). */
+  transparency?: any;
```
Comment (jgravois) [inline packages/arcgis-rest-common-types/src/webmap.ts]: `number`?

File: packages/arcgis-rest-common-types/src/webmap.ts [hunk 14]
```diff
@@ -0,0 +1,1225 @@
...
+}
+
+export interface ILayerDefinition extends IHasZM {
+  /** Boolean value indicating whether the geometry of the features in the layer can be edited. */
+  allowGeometryUpdates?: any;
```
Comment (jgravois) [inline packages/arcgis-rest-common-types/src/webmap.ts]: `boolean`

File: packages/arcgis-rest-common-types/src/webmap.ts [hunk 15]
```diff
@@ -0,0 +1,1225 @@
...
+  /** A comma separated list of supported capabilities, e.g. Query,Editing. */
+  capabilities?: string;
+  /** String value for the copyright text information for the layer. */
+  copyrightText?: any;
```
Comment (jgravois) [inline packages/arcgis-rest-common-types/src/webmap.ts]: `string`

File: packages/arcgis-rest-common-types/src/webmap.ts [hunk 16]
```diff
@@ -0,0 +1,1225 @@
...
+  layerType: "OpenStreetMap";
+  /**
+   * @deprecated
+   */
+  type?: "OpenStreetMap";
+}
+
+export interface ITiledImageServiceLayer extends ILayer {
+  /** Applicable if used as a baseMapLayer. A boolean value indicating whether or not the baseMapLayer draws on top (true) of other layers, including operationalLayers , or below (false). */
+  isReference?: any;
```
Comment (jgravois) [inline packages/arcgis-rest-common-types/src/webmap.ts]: `boolean`

**Comment by JeffJacobson** [comment]
@jgravois, I agree that those are changes that should be made. (If you're going to merge the changes into the main (Esri) repository then you don't need to worry about merging the changes into my fork.)

**Pull request merged**